crates/ibc/src/applications/transfer/context.rs

```diff
@@ -134,12 +135,9 @@ pub fn on_chan_open_init_validate(
         });
     }
 
-    if !version.is_empty() && version != &Version::new(VERSION.to_string()) {
-        return Err(TokenTransferError::InvalidVersion {
-            expect_version: Version::new(VERSION.to_string()),
-            got_version: version.clone(),
-        });
-    }
+    version
+        .verify_version_supported(Version::new(VERSION.to_string()))
+        .map_err(ContextError::from)?;
```

Comment:
This is a different check (i.e. introduces a bug). Previously, an empty version is allowed; now it is not

Reply:
👌🏻 [46b9338](https://github.com/cosmos/ibc-rs/pull/626/commits/46b933883fa15efdec087561c29c014c7ef37b2d)